### DISCUSSION TITLE
Brainstorming - Post Your Ideas

### DISCUSSION THREAD

**Discussion by Kyuhisan** [CryptechTest/cryptech#43] [Ideas]
Discussion section for collecting ideas.

**Comment by SeqSEE** [comment]
An idea for the "layers" and how the map space should be used.

### Deep Space
Contains vacuum of space. At the lower most bound of Deep Space is a impermeable layer of  barrier.  Deep Space is made up a grid of sections which players can navigate their ships to. Wormholes periodically open up randomly in Deep Space which allow for travel to and from `Void Space`. Deep Space can only be reached using special ship structures to travel in the FTL lanes or by wormholes from `Void Space`.  Asteroids  materials generate here naturally while no players are present in the section, however no atmos is generated in Deep Space.

### Red Sky Space
Contains vacuum of space. Asteroids with atmos and materials generate here naturally.

### Mid Space
Contains vacuum of space. Asteroids with atmos and materials generate here naturally.

### Space
Contains vacuum of space.. Asteroids with atmos and materials generate here naturally. Spawn is ideally on the edge of space close to the `Upper Atmosphere` at a height of around 4500.

### Upper Atmosphere
Contains air. Outer most layer of the atmosphere, the only difference from the other planetary layers is that the skybox differs and no materials generate naturally here

### Lower Atmosphere
Contains air.  Inner most layer of the atmosphere. Some materials generate here, but above around 128 is the limit of worldgen

### Planetary Crust
Contains air.  At it's maximum depth of -11000 is an impermeable layer of bedrock. 

### Gameplay Buffer Zone
Contains vacuum of space. An exclusionary zone used for game mechanics when a player needs to be in an area. Playable areas are generally generated on the fly and consist of destructible and/or indestructible nodes. 

### Void Space
Contains vacuum of space. The furthest section of space only reachable by first reaching `Deep Space` and navigating through a series of FTL lanes and wormholes and only escaped by traveling back through deep space or failing into the void to death.

![cryptech_test](https://user-images.githubusercontent.com/15011102/229680359-8f859a10-f12d-4525-ab79-16e3e97ecf00.png)

**Reply by SeqSEE** [reply]
`Deep Space` and `Gameplay Buffer Zone` are both split into 3d grids (12x2x12) consisting of approximately 5k x 5k x 5k. This way `Deep Space` can be used for ship travel and exploration and to get to `Void Space` via wormhole.

`Gameplay Buffer Zone` will be split into a similar grid so that players can be moved to it when gameplay requires some isolated area from regular gameplay (bossfights, simulations, "tardis-like" functionality, minigames).

**Reply by SeqSEE** [reply]
![cryptech_test3](https://user-images.githubusercontent.com/15011102/235786867-03167219-d098-442a-b6eb-b8561935ae8d.png)


This is the updated diagram of the map

**Comment by Naenshiko** [comment]
Main and Secondary Quests with a story about the server.

It is not mandatory to do them just for fun and with some rewards the more you advance in them.

Quests can be placed from time to time.

----------------

Economy based on mining similar to Minecraft but with a new currency just for the game.

----------------

Using the economy system and quests place POI in places on the map in which they can be traded for different items of low or high value. Taking advantage of the fact that the map can be fully explored given its size.

**Reply by SeqSEE** [reply]
We will be using the NPCs to allow server admins to setup quests. They won't be critical to gameplay (you can progress without it).

I'd also like to add some rewards to the awards mod, in game currency. 

We'd like to setup a fake blockchain which allows players to construct nodes which allow the player to "mine" the in game currency. 

Adding this and also adding some server run shops which allow players to buy and sell commodities using the in game currency would be really good.

**Reply by Boehmerang** [reply]
To add on to the Quests Idea/plans; would it be possible to make achievements with a community achievement board? This would give a reward in a way that has minimal effect on overall gameplay, but still hits the feels with a bit of community recognition for your accomplishment, and drive competition in a way of getting your name on the board. This could also be used to track who has completed which quest and help prevent quest over use exploits.

**Reply by SeqSEE** [reply]
> To add on to the Quests Idea/plans; would it be possible to make achievements with a community achievement board? This would give a reward in a way that has minimal effect on overall gameplay, but still hits the feels with a bit of community recognition for your accomplishment, and drive competition in a way of getting your name on the board. This could also be used to track who has completed which quest and help prevent quest over use exploits.

The `mobs_npc`  and `simple_dialog` mods should allow for creating, completing and tracking who's completed a quest and providing rewards, and the `awards`  mod will give any custom achievements, including ones which might only be gained by doing a quest. 

Since all this stuff will be in a database, we could make an in game and web based community board by making a mod which is dependent on the `awards` mod we are using.

**Comment by SeqSEE** [comment]
# Test Coin

Test Coin is a fake hybrid PoW/PoS cryptocurrency used as in-game currency in CryptechTest game. Test Coin can be "mined" in game by building a `PoS Miner`, `PoW Miner` or `ASIC PoW Miner` and providing lv  or mv power to it respectively.  Test Coin can take both digital and physical forms, and can be tipped digitally or rained physically with in game commands. Players can redeem their Test Coin for Metrix Coin using web3 and a Metrix Coin address they have set in game. Players can use them to trade with NPCs and NPC traders as well as in server shop. 

- As in game currency (shops, NPCs)
- Wallet in inventory
- Exchange for MRX


## Mining Nodes
### PoS Miner
- Requires LV power
- Deposit Test Coin and get rewards at a fixed rate over time
### PoW Miner
- Requires LV power
### ASIC PoW Miner
- Requires MV power
- Improved hash rate over the `PoW Miner`

**Comment by Kyuhisan** [comment]
<h1>Gameplay in regards to spaceships and exploration</h1>
This is a broad overlook at how the main gameplay loop could look like.

Player upgrades his spaceship in order to explore further away lands as he progresses though lands.

1. you start on space station that orbits somewhere in lower space near to ground (spawn area, common ground for all players like international space station)
2. you travel down to the planetary crust using jetpack (personal travel) / lander(for transporting items etc...)
3. you gather stuff, materials, liquids, gasses, fuel whatever in order to build your first spaceship, which when deployed orbits somewhere around the same level as spawn area
4. this space ship now becomes your base and you can walk around on it, maintain it etc....
5. you would upgrade this ship to expand your traveling/exploration capabilities (upgrading thrusters, boosters, navigation, fast travel etc...) (every upgrade would visually change the spaceship as well, like add bigger thrusters)
6. you could also upgrade the size, storage, plots on the ship that allow you to build other machines inside the ship like miners or oxygen generators, fuel generators etc... which would self sustain the ship instead of you having to return to ground for fuel and oxygen tanks
7. but in order to upgrade your ship further and further you would need to explore the space more and more and search for materials only available at higher altitudes
8. you could teleport your space station around, but would still have to use other means of transportation to move items from and to that space station

<h2>Means of Transportation</h2>

- Player would use different non-static forms of transportation for traveling over land and between ground and his spaceship.
- For example player could use jetpacks for personal travel or landers to transport materials from ground to spaceship.
- Could also use trains and other vehicles for transporting resources over ground (Factorio vehicles for example).

<h2>Modular Spaceships</h2>

- Each player would start with same beginner spaceship which he would then upgrade and customize to his liking. 
- He would upgrade the ship itself (stuff like thrusters, engines etc...) - STUFF THAT IS ESSENTIAL FOR THE SHIP. 
- And he would upgrade the customizable expandable part of the ship which includes plots for multiblock machines, generators, recyclers etc... - STUFF THAT IS OPTIONAL.
- Spaceships could be made into NFTs as well since each would be unique in some way or something along these lines.

<h2>Maintenance & Material Supply</h2>

- Spaceships would require maintenance materials in order to operate (like spare parts, air, fuel, electricity, water) which player would have to supply manually early on, but as he upgrades his spaceship he can implement different multiblock machines that generate/recycle these and make the spaceship self-sufficient.
- Space traveling (teleporting your spaceship) would require additional materials and use more of the maintenance materials.
- Running low or out of Air, Electricity or Water would make the ship uninhabitable, while running out of Fuel would prevent travel or running of machines. 
- There should be a good UI for this.

**Reply by SeqSEE** [reply]
I really like these ideas. The one critique I have with  the spaceships is that the player should not start with any ship, they should have to build the same starter ship from materials they found on the planet.

I can add some mods for ground transport (like trains and stuff) so we can experiment with them a bit.

**Reply by squidicuzz** [reply]
Ya this sums it up well

**Reply by SeqSEE** [reply]
Ships could be split into tiers, where each player would start by building the first tier, upgrading it within that tier and then upgrading it to the next tier. Things like other machines for the same voltage provided by the tier as well as battery boxes and solar panels can be added by the owner of the ship. The player is must gather materials in order to upgrade and maintain their ship, energy and air.

## Ship Tiers
### Scout
- LV main power system and Battery Boxes
- LV Air Handler, fans and duct system
- LV FTL Jump Drive
- Small cargo area

### Surveyor
- LV main power system and Battery Boxes
- LV Air Handler, fans and vent system
- Advanced navigation system
- LV FTL Jump Drive
- Small cargo area

###  Cruiser
- LV main power system and Battery Boxes
- LV Air Handler, fans and vent system
- Advanced navigation system
- LV FTL Jump Drive
- Medium cargo area

### Medium Cruiser
- MV main power system and Battery Boxes
- LV power subsystem
- MV Air Handler, fans and vent system
- Advanced navigation system
- MV FTL Jump Drive
- Medium cargo area

### Large Cruiser
- HV main power system and Battery Boxes
- MV power subsystem
- LV power subsystem
- MV Air Handler, fans and vent system
- Advanced navigation system
- HV FTL Jump Drive
- Medium cargo area

### Super Cruiser
- HV main power system and Battery Boxes
- MV power subsystem
- LV power subsystem
- MV Air Handler, fans and vent system
- Advanced navigation system
- HV FTL Jump Drive
- Large cargo area

### Mega Cruiser
- Nuclear reactor
- HV main power system and Battery Boxes
- MV power subsystem
- LV power subsystem
- MV Air Handler, fans and vent system
- Advanced navigation system
- HV FTL Jump Drive
- Huge cargo area

**Comment by Boehmerang** [comment]
Idea on new player spawn-ins: 
Allow players to preregister and setup a back story for their character on the website. Choices made here could change where you spawn/what you spawn with. 
Example choices: 

- Crewman that quit: left on space station with a few things, 
- Escaped Prisoner: crash on planet in escape pod with nothing, 
- Mayday Pilot: abandoned ship during malfunction (start parachuting down to planet by spawned wreckage of ship with random mats. 
- War Deserter: having snuck onto a passing merchant ship you start on space station with a fire arm, ammo, and food.
- Traveling Merchant: fled on escape pod during a pirate attack on ship; start on planet with some money, and a low tier weapon.

More back stories could even become something the community could write and vote for, allowing us to add more over time. I really like the idea of having 'universe lore' and making it seem like there is a lot more activity going on beyond our small (huge to us compared to what we're used to) part of the galaxy.

**Reply by SeqSEE** [reply]
I think this is a cool idea as long as it is optional. If it is mandatory it may become a point of friction to getting more players.

**Comment by squidicuzz** [comment]
I drew up an outline overview for how deepspace shall work..
![cryptech_space](https://user-images.githubusercontent.com/1578496/234861665-62fd281d-9d77-450b-8de9-d3dda41d7d27.png)
